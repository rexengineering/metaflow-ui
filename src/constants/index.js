export const BUSY = "BUSY";
export const OFFLINE = "OFFLINE";
export const ONLINE = "ONLINE";

export const ActionItemDeleteType = "ActionDeleteType";
export const ActionItemCheckType = "ActionCheckType";
export const ActionItemDisplayOnlyType = "ActionDisplayOnlyType";

export const InitialButtonState = 'initial';
export const LoadingButtonState = 'loading';
export const talkTrackIdentifierProp = 'iid';
export const talkTrackType = 'talktrack';