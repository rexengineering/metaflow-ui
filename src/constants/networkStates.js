export const INITIAL = "INITIAL";
export const REQUEST = "REQUEST";
export const SUCCESS = "SUCCESS";
export const FAILURE = "FAILURE";

export default {
  INITIAL,
  REQUEST,
  SUCCESS,
  FAILURE
}
