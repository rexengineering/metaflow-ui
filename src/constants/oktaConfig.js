const config = {
  oidc: {
    clientId: window.REX.OKTA.CLIENT_ID,
    issuer: window.REX.OKTA.ISSUER,
    redirectUri: window.REX.OKTA.REDIRECT_URI,
    scopes: ["openid", "profile", "email"],
    pkce: true,
    disableHttpsCheck: window.REX.OKTA.OKTA_TESTING_DISABLEHTTPSCHECK,
  },
  resourceServer: {
    messagesUrl: "http://localhost:5000/api/messages",
  },
};

export default config;
