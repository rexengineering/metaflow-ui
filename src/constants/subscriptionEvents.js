export const START_WORKFLOW = "START_WORKFLOW";
export const UPDATE_WORKFLOW = "UPDATE_WORKFLOW";
export const FINISH_WORKFLOW = "FINISH_WORKFLOW";
export const START_TASK = "START_TASK";
export const UPDATE_TASK = "UPDATE_TASK";
export const FINISH_TASK = "FINISH_TASK";
export const START_BROADCAST = "START_BROADCAST";
export const FINISH_BROADCAST = "FINISH_BROADCAST";
export const ERROR_BROADCAST = "ERROR_BROADCAST";
export const KEEP_ALIVE_SUCCESS = "SUCCESS";
export const KEEP_ALIVE_FAILURE = "FAILURE";