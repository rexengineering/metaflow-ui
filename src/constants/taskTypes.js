import { string, number, boolean } from "yup";

// Inputs
export const TEXT = "TEXT";
export const PHONE_NUMBER = "PHONE_NUMBER";
export const PERCENTAGE = "PERCENTAGE";
export const CURRENCY = "CURRENCY";
export const BOOLEAN = "BOOLEAN";
export const INTEGER = "INTEGER";
export const FLOAT = "FLOAT";

export function isInputType(field) {
  return [
    TEXT,
    PHONE_NUMBER,
    PERCENTAGE,
    CURRENCY,
    BOOLEAN,
    INTEGER,
    FLOAT,
  ].includes(field);
}

// Info
export const TABLE = "TABLE";
export const COPY = "COPY";
export const WORKFLOW_INSTANTIATOR = "WORKFLOW";

export function isInfoType(type) {
  return [TABLE, COPY, WORKFLOW_INSTANTIATOR].includes(type);
}

const numberValidationMessage = "Please enter a number";

export const validationSchemaMapping = {
  TEXT: string("Please enter a value"),
  PHONE_NUMBER: string("Please enter a value").length(10),
  PERCENTAGE: number(numberValidationMessage).typeError(
    numberValidationMessage
  ),
  CURRENCY: number(numberValidationMessage).typeError(numberValidationMessage),
  BOOLEAN: boolean(),
  INTEGER: number(numberValidationMessage)
    .typeError(numberValidationMessage)
    .integer(),
  FLOAT: number(numberValidationMessage).typeError(numberValidationMessage),
};

const taskTypes = {
  TEXT,
  PHONE_NUMBER,
  PERCENTAGE,
  CURRENCY,
  BOOLEAN,
  INTEGER,
  FLOAT,
  TABLE,
  COPY,
  WORKFLOW_INSTANTIATOR,
};

export default taskTypes;
