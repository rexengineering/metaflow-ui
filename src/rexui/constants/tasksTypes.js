import { string, number, boolean } from "yup";


const numberValidationMessage = "Please enter a number";

export const TEXT = "TEXT";
export const PHONE_NUMBER = "PHONE_NUMBER";
export const PERCENTAGE = "PERCENTAGE";
export const CURRENCY = "CURRENCY";
export const BOOLEAN = "BOOLEAN";
export const INTEGER = "INTEGER";
export const FLOAT = "FLOAT";
export const TABLE = "TABLE";
export const COPY = "COPY";
export const WORKFLOW_INSTANTIATOR = "WORKFLOW";

export const validationSchemaMapping = {
    TEXT: string("Please enter a value"),
    PHONE_NUMBER: string("Please enter a value").length(10),
    PERCENTAGE: number(numberValidationMessage).typeError(numberValidationMessage),
    CURRENCY: number(numberValidationMessage).typeError(numberValidationMessage),
    BOOLEAN: boolean(),
    INTEGER: number(numberValidationMessage)
        .typeError(numberValidationMessage)
        .integer(),
    FLOAT: number(numberValidationMessage).typeError(numberValidationMessage),
};

export function isInfoType(type) {
    return [TABLE, COPY, WORKFLOW_INSTANTIATOR].includes(type);
}

export function isInputType(field) {
    return [
        TEXT,
        PHONE_NUMBER,
        PERCENTAGE,
        CURRENCY,
        BOOLEAN,
        INTEGER,
        FLOAT,
    ].includes(field);
}