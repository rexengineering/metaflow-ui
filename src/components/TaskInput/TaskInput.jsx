import { TextField } from "@material-ui/core";
import React from "react";
import PropTypes from "prop-types";
import { TaskTypes } from "../../constants";

function TaskInput({
  dataId,
  type,
  label,
  value,
  error,
  onChange,
  helperText,
}) {
  switch (type) {
    case TaskTypes.text: {
      return (
        <TextField
          label={label}
          variant="filled"
          name={dataId}
          value={value}
          onChange={onChange}
          error={error}
          helperText={helperText}
        />
      );
    }
    default:
      return null;
  }
}

TaskInput.defaultProps = {
  label: "",
  value: "",
  dataId: "",
  type: "",
  helperText: "",
  error: false,
  onChange: () => {},
};

TaskInput.propTypes = {
  dataId: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  error: PropTypes.bool,
  onChange: PropTypes.func,
  helperText: PropTypes.string,
  label: PropTypes.string,
};

export default TaskInput;
