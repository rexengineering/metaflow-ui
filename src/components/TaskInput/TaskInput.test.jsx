import React from "react";
import { render, screen } from "@testing-library/react";
import TaskInput from "./TaskInput";
import { TaskTypes } from "../../constants";

describe("<TaskInput />", () => {
  const renderTaskInput = (
    label = "Some label",
    value = "Hello",
    type = TaskTypes.text,
    dataId = "756756"
  ) => <TaskInput label={label} value={value} type={type} dataId={dataId} />;

  it("should render", () => {
    const { asFragment } = render(renderTaskInput());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderTaskInput());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render a label", () => {
    const label = "Label!";
    render(renderTaskInput(label));
    const labelNode = screen.getByText(label);

    expect(labelNode).toBeInTheDocument();
  });

  it("should render a proper TaskInput control", () => {
    render(renderTaskInput());
    const textNode = screen.getByRole("textbox");
    expect(textNode).toBeInTheDocument();
  });
});
