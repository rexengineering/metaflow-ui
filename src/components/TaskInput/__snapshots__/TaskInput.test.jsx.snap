// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskInput /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
      data-shrink="true"
    >
      Some label
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input"
        name="756756"
        type="text"
        value="Hello"
      />
    </div>
  </div>
</DocumentFragment>
`;
